### PR TITLE
Test-proxy container + container ci.yml

### PR DESCRIPTION
**Pull request by scbedd**

### CODE REVIEW

**Comment by scbedd** [comment]
Facing some major issues with https that I can't seem to get untangled. Blocking merge until I get it working with the `examples`.

**Review by weshaggard — APPROVED** [review]
The changes look reasonable to me.

File: eng/containers/ci.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+steps:
+- ${{ each config in variables.DockerDeployments }}:
+  - task: Docker@2
+    displayName: Build ${{ config.name }} Docker Image
+    inputs:
+      command: build
+      Dockerfile: ${{ config.dockerFile }}
+      tags: $(imageTag)
+      arguments: '-t $(containerRegistry).azurecr.io/${{ config.dockerRepo }}:$(imageTag)'
```
Comment (mitchdenny) [inline eng/containers/ci.yml:32]: Won't this require whoever wants to use the test proxy to have access to the ACR instance? How will this work in an OSS context?

File: tools/test-proxy/dockerfile [hunk 2]
```diff
@@ -0,0 +1,42 @@
+FROM mcr.microsoft.com/dotnet/sdk:3.1-focal AS build
```
[the remark this answers is not in the document]
Comment (scbedd) [reply tools/test-proxy/dockerfile:1]: We're goin there!

File: eng/containers/ci.yml (hunk 1 above)
Comment (scbedd) [reply eng/containers/ci.yml:32]: This is a very good point. Thanks for bringing it up. This is a good place to start though  :)

**Pull request merged**